implementation/src/main/java/io/smallrye/config/TracedPropertiesConfigSource.java

```diff
@@ -0,0 +1,47 @@
+package io.smallrye.config;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import io.smallrye.config.common.MapBackedConfigSource;
+import io.smallrye.config.common.utils.ConfigSourceUtil;
+
+public class TracedPropertiesConfigSource extends MapBackedConfigSource implements TracedConfigSource {
+    private static final long serialVersionUID = 3969193230016229478L;
+
+    private static final String NAME_PREFIX = "TracedPropertiesConfigSource[source=";
+
+    private final URL url;
+    private final Map<String, Integer> tracedProperties = new HashMap<>();
```

Comment:
Using a second Map with boxed Integer objects could add an overhead of as much as 100 or more bytes per configuration property, which doesn't seem great.

Reply:
I know, this is not meant to be a final solution, just something to have a look. This was just to have minimal work on implementation, or I would have to rewrite Properties class all together :)